react-task/src/components/Select.jsx

```diff
@@ -12,9 +12,22 @@ const Select = ({ options }) => {
     [options]
   );
 
+  const handleOnChange = (event) => {
+    const target = event.target.name;
+    const value = event.target.value;
+    handleDataChange((prevValue) => ({ ...prevValue, [target]: value }));
+  };
+
   return (
     <div class="form-input mb-3">
-      <select class="form-select" aria-label="Select school">
+      <select
+        class="form-select"
+        aria-label="Select school"
+        value={value}
+        name={name}
+        onChange={(event) => handleOnChange(event)}
```

Comment:
You can update this as follows. Adding the enclosing function is redundant and we can just use the function reference

```suggestion
        onChange={handleOnChange}
```